tor_core/helpers.py

```diff
@@ -267,8 +294,11 @@ def run_until_dead(func, exceptions=default_exceptions):
         issues) but they can be overridden with a passed-in set.
     :return: None.
     """
+    # handler for CTRL+C
+    signal.signal(signal.SIGINT, signal_handler)
```

Comment:
This overrides a sigint signal catcher that the bot-specific code (e.g., tor_archivist) might specify, but I can live with that for now.